packages/webapp/src/containers/Finances/TransactionFilter/index.jsx

```diff
@@ -0,0 +1,72 @@
+/*
+ *  Copyright 2023 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch, useSelector } from 'react-redux';
+import FilterButton from '../../../components/Filter/FilterButton';
+import Button from '../../../components/Form/Button';
+import ModalComponent from '../../../components/Modals/ModalComponent/v2';
+import TransactionFilterContent from '../../Filter/Transactions';
+import {
+  isFilterCurrentlyActiveSelector,
+  setTransactionsFilter,
+  transactionsFilterSelector,
+} from '../../filterSlice';
+import styles from './styles.module.scss';
+
+const TransactionFilter = () => {
+  const { t } = useTranslation();
+  const [isFilterOpen, setIsFilterOpen] = useState(false);
+  const [isDirty, setIsDirty] = useState(false);
+  const isFilterActive = useSelector(isFilterCurrentlyActiveSelector('transactions'));
+
+  const transactionsFilter = useSelector(transactionsFilterSelector);
+  const dispatch = useDispatch();
+
+  const handleApply = () => {
+    dispatch(setTransactionsFilter(filterRef.current));
+    setIsFilterOpen(false);
+  };
```

Comment:
Could we reset the filter to the initial state when all filters have been selected?

```suggestion
  const handleApply = () => {
    const allAreActive = () =>
      Object.values(filterRef.current).every((filter) =>
        Object.values(filter).every((subFilter) => subFilter.active),
      );

    if (allAreActive()) {
      dispatch(resetTransactionsFilter());
    } else {
      dispatch(setTransactionsFilter(filterRef.current));
    }
    setIsFilterOpen(false);
  };
```

Reply:
I ended up baking this in in the `isFilterCurrentlyActiveSelector` which should also take care of showing the filter as active now if all filters are cleared 🙂